radixdlt/src/main/java/com/radixdlt/middleware2/converters/AtomToBinaryConverter.java

```diff
@@ -29,19 +29,19 @@ public AtomToBinaryConverter(Serialization serializer) {
 		this.serializer = serializer;
 	}
 
-	public byte[] toLedgerEntryContent(Atom atom) {
+	public byte[] toLedgerEntryContent(LedgerAtom atom) {
 		try {
 			return serializer.toDson(atom, DsonOutput.Output.PERSIST);
 		} catch (SerializationException e) {
 			throw new RuntimeException(String.format("Serialization for Atom with ID: %s failed", atom.getAID()));
 		}
 	}
 
-	public Atom toAtom(byte[] ledgerEntryContent) {
+	public LedgerAtom toAtom(byte[] ledgerEntryContent) {
 		try {
-			return serializer.fromDson(ledgerEntryContent, Atom.class);
+			return serializer.fromDson(ledgerEntryContent, LedgerAtom.class);
 		} catch (SerializationException e) {
-			throw new RuntimeException("Deserialization of Atom failed");
+			throw new RuntimeException("Deserialization of Atom failed", e);
```

Comment:
Kudos for exception chaining, but really should be an IllegalStateException, rather than raw RuntimeException.